numba/cuda/simulator/cudadrv/devicearray.py

```diff
@@ -82,6 +82,8 @@ def __setitem__(self, idx, val):
     def copy_to_host(self, ary=None, stream=0):
         if ary is None:
             ary = np.empty_like(self._ary)
+        else:
+            check_array_compatibility(self, ary)
         # NOTE: np.copyto() introduced in Numpy 1.7
```

Comment:
Does numba still support numpy<1.7? CI tests only go back to 1.9.

Reply:
We should actively drop any workarounds for NumPy < 1.9 at this point.